operator/src/main/java/org/keycloak/operator/controllers/KeycloakUpdateJobDependentResource.java

```diff
@@ -218,8 +219,14 @@ private static List<String> replaceStartWithUpdateCommand(List<String> currentAr
         return Stream.concat(updateArgs.stream(), currentArgs.stream().filter(arg -> !arg.equals("start"))).toList();
     }
 
-    private static String keycloakHash(Keycloak keycloak) {
-        return Utils.hash(List.of(keycloak.getSpec()));
+    static String keycloakHash(Keycloak keycloak) {
+        return Utils.hash(
+                List.of(new KeycloakSpecBuilder(keycloak.getSpec()).withInstances(null).withLivenessProbeSpec(null)
+                        .withHostnameSpec(null).withStartOptimized(null).withTracingSpec(null).withTruststores(null)
+                        .withHttpManagementSpec(null).withProxySpec(null).withStartupProbeSpec(null)
+                        .withReadinessProbeSpec(null).withResourceRequirements(null).withSchedulingSpec(null)
+                        .withBootstrapAdminSpec(null).withNetworkPolicySpec(null).withIngressSpec(null)
+                        .withImagePullSecrets().withImportSpec(null).withUpdateSpec(null).build()));
```

Comment:
The following spec may be ignored too:
* TransactionSpec
* ProxySpec

Reply:
ProxySpec was already covered. Added TransactionSpec.